File: digivise/src/main/java/org/nevmock/digivise/infrastructure/config/mongo/MongoConfig.java
package org.nevmock.digivise.infrastructure.config.mongo;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.convert.converter.Converter;
import org.springframework.data.mongodb.core.convert.MongoCustomConversions;

import java.util.ArrayList;
import java.util.List;

@Configuration
public class MongoConfig {
    @Bean
    public MongoCustomConversions customConversions() {
        List<Converter<?,?>> convs = new ArrayList<>();
        convs.add(new DocumentToProductKeywordResponseDtoConverter());
        return new MongoCustomConversions(convs);
    }
}

File: digivise/src/main/java/org/nevmock/digivise/infrastructure/config/mongo/MongoConfig.kt
package org.nevmock.digivise.infrastructure.config.mongo

//@Configuration
//open class MongoConfig {
//
//    @Value("\${mongodb.uri}")
//    private lateinit var mongoUri: String
//
//    @Bean
//    open fun mongoClient() : MongoClient {
//        val connectionString = ConnectionString(mongoUri)
//
//        val settings = MongoClientSettings.builder()
//            .applyConnectionString(connectionString)
//            .build()
//
//        return MongoClients.create(settings)
//    }
//}